### ISSUE TITLE
Rate Limit on "Send Extension PubSub Message"

### ISSUE THREAD

**Issue by BarryCarlyon** [twitchdev/issues#612]
**Brief description**

https://dev.twitch.tv/docs/api/reference#send-extension-pubsub-message

I feel the rate limit documetnation on "Send Extension PubSub Message" is not a good representation of the rate limit

> Extension PubSub has a rate limit of 100 requests per minute for a combination of Extension client ID and broadcaster ID.

It should read more similar to the behaviour which also leads developers to write better PubSub usage.

And also add a note here about the size of a message

**Expected documentation**

Extension PubSub has a maximum size of 5kb per message.
Extension PubSub has a rate limit of 1 message per second.
Within a given 60 seconds period you have 40 additional rate limit points to spend.

The Global topic is a seperate rate limit pool to the Channel topic.
Whispers use the channel topic for the channelID that the whisper is routed via.

or based on testing the behaviour

The rate limit is 100 points with a refill of 1 point per second.

**Additional context or questions**

Basically the rate limit documentation on the endpoint don't give justice to the actual rate limit that is on production.
Which also seems to be different to non helix/deprecated version of this endpoint.

Old:
1 per second with 40 left at the end of a minute and it refrills to full at the end of the minute

New:
refills at 1 per second with a start pool of 100

**Comment by BarryCarlyon** [comment]
Even then it's not _quite_ 1 per second
